### PR TITLE
Introduce reference package for a Google Cloud SQL service instance

### PR DESCRIPTION
**Pull request by hoegaarden**
Note: package repo bundle not updated yet, thus not "released"

### CODE REVIEW

File: bundles/google/config-connector/cloudsql/bundle/config/00-schema.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+version: POSTGRES_14
+
+#@schema/title "the region this Cloud SQL instance should be deployed into"
+region: europe-west6
+
+#@schema/title "the machine instance type for the Cloud SQL instance"
+tier: db-g1-small
+
+#@schema/title "a set of labels which should be applied to all resources related to this instance"
+labels:
```
[the remark this answers is not in the document]
Comment (Samze) [reply bundles/google/config-connector/cloudsql/bundle/config/00-schema.yml:29]: Yeah makes sense, let's expose that claimable label too.

**Comment by hoegaarden** [comment]
In my local tests I stumbled upon an issue with the secretgen-controller. It seems if you use a password to create a secret, once it is created, this secret is not reconciled anymore. This is not blocking this PR or anything, but an interesting find.

- [slack](https://kubernetes.slack.com/archives/CH8KCCKA5/p1657900918552639)
- [issue](https://github.com/vmware-tanzu/carvel-secretgen-controller/issues/91)

**Comment by gmrodgers** [comment]
Built this locally and deployed and created, a GCP CloudSQL instance 👍 :

```console
kapp inspect -a first-ctrl -n service-instances
Target cluster 'https://35.199.34.76' (nodes: gke-docs-43-test-default-pool-5e75099b-26rn, 5+)

Resources in app 'first-ctrl'

Namespace          Name               Kind            Owner    Rs  Ri  Age  
service-instances  first              SQLDatabase     kapp     ok  -   7m  
^                  first              SQLInstance     kapp     ok  -   7m  
^                  first              SQLUser         kapp     ok  -   7m  
^                  first              SecretTemplate  kapp     ok  -   7m  
^                  first-admin-creds  Password        kapp     ok  -   7m  
^                  first-admin-creds  Secret          cluster  ok  -   7m  
^                  first-obj-reader   Role            kapp     ok  -   7m  
^                  first-obj-reader   RoleBinding     kapp     ok  -   7m  
^                  first-obj-reader   ServiceAccount  kapp     ok  -   7m  
^                  first-user-creds   Password        kapp     ok  -   7m  
^                  first-user-creds   Secret          cluster  ok  -   7m  

Rs: Reconcile state
Ri: Reconcile information

11 resources

Succeeded
```

File: bundles/google/config-connector/cloudsql/bundle/config/99-kapp-config.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+---
+apiVersion: kapp.k14s.io/v1alpha1
```
[the remark this answers is not in the document]
Comment (Samze) [reply bundles/google/config-connector/cloudsql/bundle/config/99-kapp-config.yml:2]: Looks better! I will open an issue tomorrow to port some of these general improvements to the RDS package

**Comment by Samze** [comment]
Thanks for addressing the feedback! LGTM.

**Pull request merged**